src/foam/core/FObject.js

```diff
@@ -808,6 +808,12 @@ foam.CLASS({
 
     function compareTo(other) {
       if ( other === this ) return 0;
+
+      // Use primitive type comparator if available
+      if (foam.typeOf(other).id !== "foam.core.FObject") {
```

Comment:
There might be an easier way to check for primitive types... but this was the shortest solution that occurred to me